### PR TITLE
more def improvements

### PR DESCRIPTION
**Pull request by rmccallum81**
More improvements but I have a comments below to help clarify

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ declare namespace Moleculer {
 		warn: (...args: any[]) => void;
 		info: (...args: any[]) => void;
 		debug?: (...args: any[]) => void;
-		trace: (...args: any[]) => void;
+		trace?: (...args: any[]) => void;
```
Comment (rmccallum81) [inline index.d.ts]: This is to make it compatible with `console` in node

File: index.d.ts [hunk 2]
```diff
@@ -58,18 +66,19 @@ declare namespace Moleculer {
...
-	interface ServiceActionSchema {
-		cache?: boolean;
-		params?: object;
-		handler: (ctx: Context) => Promise<any>;
-	}
-
 	interface ServiceSettingSchema {
 		port?: number | string;
-		routes?: Route[];
+		routes?: RouteSchema[];
```
Comment (rmccallum81) [inline index.d.ts]: Are port and routes not specific to the moleculer-web module? If so they should be in that project as an extension of this interface.

Comment (dani8art) [reply index.d.ts]: Yes, It's true. I introduced this mistake, I thought `ServiceSettingSchema` has these two properties in `moleculer` by default. I'm sorry.

File: index.d.ts [hunk 3]
```diff
@@ -58,18 +66,19 @@ declare namespace Moleculer {
...
+		transport?: string | Transporter;
```
Comment (icebob) [inline index.d.ts]: `transport` is not a service setting as well.

File: index.d.ts (hunk 2 above)
Comment (icebob) [reply index.d.ts]: Please remove them.

File: index.d.ts (hunk 3 above)
Comment (dani8art) [reply index.d.ts]: Then, service setting has only: 
```
[name: string]: any;
```
is it true?

Comment (icebob) [reply index.d.ts]: Yes, but there are two internal optional settings: https://moleculer.services/docs/service.html#Internal-settings

**Comment by icebob** [comment]
Thanks!

**Pull request merged**